packages/nimble-components/src/patterns/dropdown/helpers.ts

```diff
@@ -0,0 +1,24 @@
+import {
+    controlHeight as controlHeightToken,
+    smallPadding
+} from '../../theme-provider/design-tokens';
+
+const maxVisibleOptions = 10.5;
```

Comment:
I guess the idea of "show an extra half item to indicate that there are more you can scroll to" breaks down when there are groups since our group items aren't the same height as our list items.

![image](https://github.com/ni/nimble/assets/10500124/86d213a8-99c5-4d3e-8e78-a893bb99ac7b)

I'm not sure what, if anything, to do about it. I don't think it's worth it (or even possible) to do calculations that factor in how many groups are visible. I think our options are either:
1. leave it as is
2. decide that the extra half item isn't worth it since it won't work for some cases, and just pick a fixed pixel height instead of doing all these calculations

Reply:
Other ideas that came up:
1. Is there some other partial size that will "generally clip" like 10.75
2. [Scroll driven animations are not available in all browsers today](https://caniuse.com/mdn-css_properties_animation-timeline_scroll) but maybe in the future we can use them to make a useful style like [this example](https://x.com/jh3yy/status/1758284287189958973) ([codepen](https://codepen.io/jh3y/pen/oNVmzby) for supported browsers)

Reply:
I think we leave it as is, since it still makes sense to pick a limit based on a number of visible options, and in the (more common?) case where options are not grouped, showing a partial item is nice.